### PR TITLE
perf(terraform-provider): use package index instead of complete index

### LINKED ISSUE #9590: Add long-term cache for requests to releases.hashicorp.com
**What would you like Renovate to be able to do?**

See: `queryReleaseBackend()` function

**Did you already have any implementation ideas?**

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Use the provider specific index for gathering versions instead of complete index. 

## Context
First find using the new tracing instrumentation 😄 

Should make longterm caching the index unnecessary:      
Closes https://github.com/renovatebot/renovate/issues/9590

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/terraform-provider/types.ts [hunk 1]
```diff
@@ -34,13 +34,10 @@ export interface TerraformProviderVersionsVersion {
...
 }
 
-export type TerraformProviderReleaseBackend = Record<
-  string,
-  {
-    name: string;
-    versions: VersionsReleaseBackend;
-  }
->;
+export type TerraformProviderReleaseBackend = {
```
Comment (viceice) [inline lib/modules/datasource/terraform-provider/types.ts:37]: please convert to interface

**Review by viceice — COMMENTED** [review]
otherwise LGTM

**Review by viceice — COMMENTED** [review]
can you do some real tests? was there an update on the hashicorp API?

**Comment by secustor** [comment]
Not, yet.    
I guess so, as I'm remembering trying to use that index during the initial implementation.

**Comment by secustor** [comment]
E2E tested with: https://github.com/secustor/renovate_terraform_lockfile/pulls

**Review by rarkins — APPROVED** [review]

**Pull request merged**